lib/puppet/http/response.rb

```diff
@@ -1,37 +1,91 @@
+#
+# @api private
+#
+# Represents the response returned from the server from an HTTP request
+#
 class Puppet::HTTP::Response
-  attr_reader :nethttp, :url
+  # @api private
+  # @return [Net::HTTP] the Net::HTTP response
+  attr_reader :nethttp
 
+  # @api private
+  # @return [URI] the response uri
+  attr_reader :url
+
+  #
+  # @api private
+  #
+  # Object to represent the response returned from an HTTP request
+  #
+  # @param [Net::HTTP] nethttp the request response
+  # @param [URI] url
+  #
   def initialize(nethttp, url)
     @nethttp = nethttp
     @url = url
   end
 
+  #
+  # @api private
+  #
+  # Extract the response code
+  #
+  # @return [Integer] Response code for the request
+  #
   def code
     @nethttp.code.to_i
   end
 
+  #
+  # @api private
+  #
+  # Extract the response message
+  #
+  # @return [String] Response message for the request
+  #
   def reason
     @nethttp.message
   end
 
+  #
+  # @api private
+  #
+  # Extract the response body
+  #
+  # @return [String] Response body for the request
+  #
   def body
     @nethttp.body
   end
 
+  #
+  # @api private
+  #
```

Comment:
Maybe mention this method streams the response body to the caller in chunks whereas the `body` method above returns the entire response body. Callers should use one or the other but not both for the same response.